core/src/main/scala/org/apache/gearpump/cluster/client/RunningApplication.scala

```diff
@@ -18,17 +18,21 @@
 package org.apache.gearpump.cluster.client
 
 import akka.pattern.ask
-import akka.actor.ActorRef
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
 import akka.util.Timeout
-import org.apache.gearpump.cluster.ClientToMaster.{ResolveAppId, ShutdownApplication}
-import org.apache.gearpump.cluster.MasterToClient.{ResolveAppIdResult, ShutdownApplicationResult}
+import org.apache.gearpump.cluster.ClientToMaster.{RegisterAppResultListener, ResolveAppId, ShutdownApplication}
+import org.apache.gearpump.cluster.MasterToClient._
+import org.apache.gearpump.cluster.client.RunningApplication.{AppResultListener, WaitUntilFinish}
 import org.apache.gearpump.util.ActorUtil
+import org.apache.gearpump.cluster.client.RunningApplication._
 
+import scala.concurrent.duration._
 import scala.concurrent.Future
 import scala.util.{Failure, Success}
 import scala.concurrent.ExecutionContext.Implicits.global
 
-class RunningApplication(val appId: Int, master: ActorRef, timeout: Timeout) {
+class RunningApplication(val appId: Int, master: ActorRef, timeout: Timeout,
```

Comment:
I think we need a user interface here. `RunningApplication` could be the default implementation. Plus, please add guide on how to use the interface.